sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2702,7 +2715,10 @@ object Sequence {
          |    $arr[$i] = ($elemType) ($t / ${scale}L);
          |    $i += 1;
          |    $t = org.apache.spark.sql.catalyst.util.DateTimeUtils.timestampAddInterval(
-         |       $startMicros, $i * $stepMonths, $i * $stepMicros, $zid);
+         |       $startMicros,
+         |       new org.apache.spark.unsafe.types.CalendarInterval(
```

Comment:
Sorry I missed this place. It has performance penalty if we create an object every time. Let's go back to the previous version. Sorry for the back and forth!

Reply:
np. I reset the changes back.